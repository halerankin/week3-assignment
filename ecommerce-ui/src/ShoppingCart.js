import React from 'react';

export default class ShoppingCart extends React.Component {

    render() {
        const totalCost = this.props.totalCost;
        const houseList = this.props.houses.map(
        (house, idx) => {
            return (
                <li key={idx}>
                    <span>{house.title}</span>
                    <button onClick={() => this.props.onRemoveHouse(idx)}>Remove</button>
                </li>
            );
        })

        return (
            <div className="wrapper shopping-cart">
                <p>Your cart: </p>
                <ul>
                   {houseList}
                </ul>
                <p>Total cost: {totalCost}</p>
                <hr />
            </div>
        );
    }
}