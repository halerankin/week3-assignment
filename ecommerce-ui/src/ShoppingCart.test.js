import React from 'react';
import {shallow} from 'enzyme';
import ShoppingCart from './ShoppingCart';

it('shallow renders', () => {
    const wrapper = shallow(<ShoppingCart />);
    expect(wrapper).toMatchSnapshot();
  });