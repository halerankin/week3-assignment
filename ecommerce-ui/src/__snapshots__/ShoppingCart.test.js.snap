// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders 1`] = `
<div
  className="wrapper shopping-cart"
>
  <p>
    Your cart: 
  </p>
  <ul />
  <p>
    Total cost: 
  </p>
  <hr />
</div>
`;
